src/index.ts

```diff
@@ -1,16 +1,84 @@
-import { Server } from '@hocuspocus/server';
+import {
+  Server,
+  Document,
+  onStoreDocumentPayload,
+  afterLoadDocumentPayload,
+} from '@hocuspocus/server';
 import { Logger } from '@hocuspocus/extension-logger';
 import { Elasticsearch } from '@cofacts/hocuspocus-extension-elasticsearch';
+import { Snapshot, addVersion } from './snapshot';
+import { yDocToProsemirrorJSON } from 'y-prosemirror';
+import { Node } from 'prosemirror-model';
+import { schema } from 'prosemirror-schema-basic';
+import elasticsearch from '@elastic/elasticsearch';
 import 'dotenv/config';
 
+const elasticsearchOpts: elasticsearch.ClientOptions = {
+  node: process.env.ELASTICSEARCH_URL,
+};
+
+const afterLoadDocument = async ({ document }: afterLoadDocumentPayload) => {
+  // migration script, for those article had transcript already but didn't have version snapshots
+  if (
+    document.getArray('versions').length === 0 &&
+    document.getXmlFragment('prosemirror').toJSON()
```

Comment:
Question: as `versions` are moved to ES index instead of document, does this condition still work as expected?

Reply:
Replaced with a migration script.